### PR TITLE
escaping - in RegExp also better gitignore

### PR DESCRIPTION
**Pull request by apolishch**

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -3,5 +3,7 @@ var assert = require('assert');
 var escapeStringRegexp = require('./');
 
 it('should escape RegExp special characters', function () {
-	assert.strictEqual(escapeStringRegexp('\\ ^ $ * + ? . ( ) | { } [ ]'), '\\\\ \\^ \\$ \\* \\+ \\? \\. \\( \\) \\| \\{ \\} \\[ \\]');
+        console.log('asshat');
```
Comment (jbnicolai) [inline test.js]: ?

**Comment by jbnicolai** [comment]
Could you explain why you think `/-/` should be ignored?

For now: closed.

File: test.js (hunk 1 above)
Comment (apolishch) [inline test.js]: My bad on the dodgy console.log, wasn't supposed to get pushed up -_-

**Comment by apolishch** [comment]
http://www.cs.cf.ac.uk/Dave/Internet/NEWS/regexp.html
A minus sign (-) within square brackets indicates a range of consecutive ASCII characters. For example, [0-9] is the same as [0123456789]. The minus sign loses its special meaning if it's the first (after an initial ^, if any) or last character in the string.

It is a special character so I want to escape it :)

**Comment by jbnicolai** [comment]
@apolishch but the square brackets will already be escaped, meaning the `-` will never mean anything but the raw character.

**Comment by apolishch** [comment]
Whelp, now I feel silly.

**Comment by sindresorhus** [comment]
> Let me know if you'd like me to resubmit with the gitignore

Nope, that should be in your own global gitignore, not here: https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

**Pull request closed without merging**